src/main/java/core/basesyntax/impl/StorageImpl.java

```
@@ -3,17 +3,61 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+
+    private K[] keys;
+    private V[] values;
+    private int size = 0;
+
+    @SuppressWarnings("unchecked")
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_SIZE];
+        values = (V[]) new Object[MAX_SIZE];
+    }
+
+    @SuppressWarnings("unchecked")
+    public StorageImpl(K key, V value) {
+        keys = (K[]) new Object[MAX_SIZE];
+        values = (V[]) new Object[MAX_SIZE];
+
+        keys[size] = key;
+        values[size] = value;
+        size++;
+    }
+
     @Override
     public void put(K key, V value) {
+        if (size == MAX_SIZE) {
+            return;
```

The `put` method currently returns immediately if the storage is full. It might be more appropriate to throw an exception or handle this case in a way that informs the caller that the storage is full, as silently failing might lead to unexpected behavior.